Sources/BraintreeVenmo/BTVenmoClient.swift

```diff
@@ -83,21 +83,23 @@ import BraintreeCore
         apiClient.sendAnalyticsEvent(BTVenmoAnalytics.tokenizeStarted, isVaultRequest: shouldVault, linkType: linkType)
         let returnURLScheme = BTAppContextSwitcher.sharedInstance._returnURLScheme
 
-        if returnURLScheme.isEmpty {
-            NSLog(
-                "%@ Venmo requires a return URL scheme to be configured via [BTAppContextSwitcher setReturnURLScheme:]",
-                BTLogLevelDescription.string(for: .critical)
-            )
-            notifyFailure(with: BTVenmoError.appNotAvailable, completion: completion)
-            return
-        } else if let bundleIdentifier = bundle.bundleIdentifier, !returnURLScheme.hasPrefix(bundleIdentifier) {
-            NSLog(
-                // swiftlint:disable:next line_length
-                "%@ Venmo requires [BTAppContextSwitcher setReturnURLScheme:] to be configured to begin with your app's bundle ID (%@). Currently, it is set to (%@)",
-                BTLogLevelDescription.string(for: .critical),
-                bundleIdentifier,
-                returnURLScheme
-            )
+        if universalLink?.absoluteString.isEmpty == true || universalLink?.absoluteString == nil {
+            if returnURLScheme.isEmpty {
+                NSLog(
+                    "%@ Venmo requires a return URL scheme or universal link to be configured.",
+                    BTLogLevelDescription.string(for: .critical)
+                )
+                notifyFailure(with: BTVenmoError.appNotAvailable, completion: completion)
+                return
+            } else if let bundleIdentifier = bundle.bundleIdentifier, !returnURLScheme.hasPrefix(bundleIdentifier) {
+                NSLog(
+                    // swiftlint:disable:next line_length
+                    "%@ Venmo requires [BTAppContextSwitcher setReturnURLScheme:] to be configured to begin with your app's bundle ID (%@). Currently, it is set to (%@)",
```

Comment:
Do we want to update this to Swift-y syntax instead of ObjC? Would you consider it a breaking change?

Reply:
Yeah I have always wondered how/if the `NSLog`s are actually useful to merchants. It's also interesting that we return an `appNotAvailable` for a missing return URL scheme and return nothing and instead fail later if the bundle ID doesn't contain the return URL scheme. To me it seems like they should both be handled the same, though not sure what we consider in scope for this PR.

Reply:
We don't allow URL schemes at all in v7 ([see code](https://github.com/braintree/braintree_ios/blob/v7/Sources/BraintreeVenmo/BTVenmoClient.swift)) so this is already cleaned up there. 😄